Sam/docs

samholmes opened the pull request:
### CHANGELOG

Does this branch warrant an entry to the CHANGELOG?

- [ ] Yes
- [x] No

### Dependencies

<!-- Replace line with PRs which this PR depends if any --> none

### Description

<!-- Describe your changes textually and/or pictorially if necessary --> none


---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1211105048261470

samholmes committed:
docs: Add comprehensive documentation for developers and AI agents

- Create AGENTS.md with build commands and code style guidelines
- Add TypeScript style guide covering imports, types, and conventions
- Document swap plugin architecture patterns and best practices
- Capture critical business rules for DEX wallet requirements
- Create step-by-step guide for adding new exchanges
- Track deprecated APIs that need migration
- Reference Edge company-wide conventions
- Update README with quick links to key documentation

This documentation provides a knowledge base for both human developers
and AI coding assistants working on the edge-exchange-plugins project.

samholmes committed:
docs: Add debug server instructions for faster development

- Document DEBUG_EXCHANGE env setting in edge-react-gui
- Present debug server as recommended development approach
- Clarify difference between debug server and direct linking methods
- Update debugging tips to emphasize hot-reload benefits

samholmes committed:
docs: Add comprehensive plugin expectations and requirements

- Document chain support mapping and parameter access patterns
- Emphasize critical fee estimation via transaction creation
- Add destination tag/memo handling requirements
- Clarify fixed vs variable quote specifications
- Document reverse quote implementation
- Expand error handling requirements with specific examples
- Add tokenId to contract address mapping explanation
- Include best practices for all new requirements

samholmes committed:
fix: Correct chain ID access and Edge type usage in docs

- Remove incorrect usage of deprecated defaultSettings for chain ID
- Clarify that pluginId uniquely identifies the network
- Update type annotations to use proper Edge types (EdgeTokenId)
- Explain that tokenId is null for native/mainnet tokens
- Document EdgeCurrencyWallet structure with currencyInfo/currencyConfig
- Emphasize pluginId as the primary chain identifier

samholmes committed:
fix: Make tokenId documentation chain-agnostic

- Remove EVM-specific references for tokenId
- Clarify tokenId applies to any chain with tokens
- Explain tokenId format varies by chain type
- Use more generic language (native currency vs tokens)

samholmes committed:
docs: Enhance swap plugin architecture documentation

- Add comprehensive plugin system fundamentals section
- Document how plugins attach to edge-core-js via factory pattern
- Explain EdgeCorePluginOptions and plugin lifecycle
- Correct API usage based on actual Edge types:
  - Fix currency code access (must be derived from wallet/tokenId)
  - Update to use networkFees array instead of networkFee
  - Correct memo API to use memos array
  - Fix EdgeSwapQuote interface properties
- Add getCurrencyCode helper function example
- Clarify factory function signatures and registration

paullinator commented on docs/guides/adding-new-exchange.md:
```diff
@@ -0,0 +1,208 @@
+    throw new SwapBelowLimitError(swapInfo, MIN_AMOUNT)
+  }
+
+  // 5. Return EdgeSwapQuote
```
API should return all errors in an array and errors should be surfaced with this priority.

1. region restriction
2. asset support
4. min/max amount

paullinator commented on docs/references/api-deprecations.md:
```diff
@@ -0,0 +1,64 @@
+
+This document tracks deprecated APIs in edge-core-js that are still used in the codebase. These should be migrated when possible.
+
+## Deprecated APIs Currently in Use
```
Use of currencyCode is deprecated as well

paullinator commented on docs/patterns/swap-plugin-architecture.md:
```diff
@@ -56,6 +56,207 @@ export const swapInfo: EdgeSwapInfo = {
+
+// For XRP, XLM, etc.
+if (tag != null) {
+  spendInfo.spendTargets[0].uniqueIdentifier = tag;
```
I think the convention is now `.memos[]`

paullinator commented on docs/patterns/swap-plugin-architecture.md:
```diff
@@ -56,6 +56,207 @@ export const swapInfo: EdgeSwapInfo = {
+
+// Insufficient liquidity
+if (response.error === "INSUFFICIENT_LIQUIDITY") {
+  throw new InsufficientLiquidityError(swapInfo, {
```
Hallucination?

paullinator commented on docs/patterns/swap-plugin-architecture.md:
```diff
@@ -56,6 +56,207 @@ export const swapInfo: EdgeSwapInfo = {
+  expirationDate?: Date; // Required for fixed quotes
+
+  // For fixed quotes
+  guaranteedAmount?: string; // The guaranteed output amount
```
Hallucination?

samholmes committed:
refactor: Replace currency code terminology with symbol mapping

- Update fetchSwapQuote to use pluginId/tokenId → symbol mapping
- Add comprehensive mapping examples for chains and tokens
- Include EVM chain ID mapping for exchanges that need it
- Emphasize API-driven validation over hardcoded values
- Fix duplicate section in prerequisites
- Update error handling to reference assets instead of currencies

samholmes committed:
docs: Document proper error throwing patterns with priority

- Add comprehensive error handling section with priority order
- Show API-driven validation instead of hardcoded values
- Document error priority: region → asset support → limits
- Update quote request pattern to use API validation
- Replace currency validation with asset validation via API

samholmes committed:
docs: Emphasize API-driven validation principle

- Add explicit section on API-driven validation
- Show BAD vs GOOD examples of hardcoded vs API validation
- Update best practices to prioritize API validation
- Clarify what exchange APIs should return
- Reinforce no hardcoded limits or asset lists

samholmes committed:
refactor: Remove remaining currency code references

- Replace currency code helpers with exchange symbol mapping
- Update error throwing examples to use request object
- Fix deprecated currencyCode field in networkFee
- Ensure consistent use of pluginId/tokenId terminology

samholmes committed:
fix(docs): replace hallucinated API response examples with real ones

- Remove fictional apiResponse properties that don't exist
- Add real examples from SideShift, ChangeNow, and LetsExchange
- Emphasize that each exchange has unique API response formats
- Show actual error handling patterns from existing plugins
- Update testing guide to reflect API-specific validation

Addresses PR feedback about 'hallucination' in documentation

paullinator commented on docs/business-rules/wallet-validation-rules.md:
```diff
@@ -0,0 +1,84 @@
+
+### DEX Same-Wallet Requirement
+
+- DEX swaps happen in a single transaction on-chain
```
This is not true. Most DEXs can have different source and dest addresses. Only the XRP DEX and the Fantom/Sonic bridge require this

paullinator commented on docs/business-rules/wallet-validation-rules.md:
```diff
@@ -0,0 +1,84 @@
+async fetchSwapQuote(request: EdgeSwapRequest): Promise<EdgeSwapQuote> {
+  // Validate same wallet requirement for DEX
+  if (request.fromWallet !== request.toWallet) {
+    throw new Error(`${swapInfo.displayName} requires same wallet for swap`)
```
There is a new swap error specifically for this. See SwapAddressError in fantomSonicUpgrade.ts
